### DISCUSSION TITLE
Text File Encoding - Support for UTF-8 and More

### DISCUSSION THREAD

**Discussion by GCuser99** [GCuser99/SeleniumVBA#32] [General]
I was developing functionality to save/read webpage source to/from file, when I encountered a couple of issues:

1) FSO defaults to saving ascii format, which breaks with (generally) non-ascii HTML.
2) Can specify Unicode "save" in FSO, but it creates UTF-16, which can break some (dumb?) applications that assume Unicode=UTF-8.

In particular, the Firefox browser json-viewer will break if input json file is saved in UTF-16 (assumes its UTF-8).

EDITED: I decided against writing code to support UTF-8 formated text files, and instead changed all of the fso.CreateTextfile calls to save to Unicode (UTF-16). I have reported the Firefox browser bug [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1797871).

**Comment by 6DiegoDiego9** [comment]
I'm incompetent in this but I'm interested in this updated feature.
Thanks for your efforts!

**Comment by 6DiegoDiego9** [comment]
Agree 100%.
_____
As you know, I now (switched recently) use Edge as my main (manual) browser.
Some days ago I tried the method WebCapabilities.SetDownloadPrefs with Edge to automate some file downloading for my company and got stuck with the fact that Edge still wants to prompt the user to press "Open / Save As". I was unable to find a command to instruct it to not prompt.
Then I tried the same method with Chrome and it downloaded the files without prompting the user, and so I came to the decision to switch to Chrome as my default choice for SeleniumVBA.
I think that having a different icon (Chrome vs Edge) in the taskbar for Selenium is a plus and I also feel safer not risking to mess up my everyday main browser Edge with my VBA experiments.

I therefore agree to limit and concentrate our (limited) efforts on Chrome.

**Comment by GCuser99** [comment]
Can you provide a non-proprietary example where the SetDownloadPrefs is not working for Edge? Or is it failing only on some proprietary application? I tested it on all three browsers and did not encounter a problem (see test_file_download in the test_FileUpDownload module). I have also tested the SetDownloadPrefs on other websites for all three browsers and not had a problem.

I'm doing the same - mostly using Chrome for my automation apps, and Edge for my manual browsing. 

I still want this code to support all three browsers mostly equally - but not to the extent that we are performing all kinds of acrobatics to accommodate browser bugs or browser-specific functionality limitations. So far keeping to that goal has not been a big effort. As you may have noticed, there are currently a few (mostly minor) Selenium functionality limitations for Firefox - the code returns a warning to users if they try to access those.

**Reply by 6DiegoDiego9** [reply]
Ahhh I just tested with test_FileUpDownload on both my company's PC and my personal PC and while the first prompts me, the second don't!
Eventually I found that my company's PC has this Edge setting "ON", greyed out and with a lock icon on its left:

![Immagine 2022-10-20 200826](https://user-images.githubusercontent.com/109476941/197025391-049dfaa6-3246-4e5a-a6a2-146625483adb.jpg)

Issue closed :)

**Comment by 6DiegoDiego9** [comment]
SetDownloadPrefs discussion continued here:  [https://github.com/GCuser99/SeleniumVBA/issues/34](https://github.com/GCuser99/SeleniumVBA/issues/34)
